test/lib/epochs.test.js

```diff
@@ -0,0 +1,129 @@
+const test = require('tape')
+const { promisify: p } = require('util')
+
+const Server = require('../helpers/testbot')
+const replicate = require('../helpers/replicate')
+
+function Run (t) {
+  // this function takes care of running a promise and logging
+  // (or testing) that it happens and any errors are handled
+  return async function run (label, promise, opts = {}) {
+    const {
+      isTest = true,
+      timer = false,
+      logError = false
+    } = opts
+
+    if (timer) console.time('> ' + label)
+    return promise
+      .then(res => {
+        if (isTest) t.pass(label)
+        else console.log(' ', label)
+        return res
+      })
+      .catch(err => {
+        t.error(err, label)
+        if (logError) console.error(err)
+      })
+      .finally(() => timer && console.timeEnd('> ' + label))
+  }
+}
+
+test('lib/epochs', async t => {
+  const run = Run(t)
+
+  const peers = [Server(), Server(), Server()]
+  t.teardown(() => peers.forEach(peer => peer.close(true)))
+
+  const [alice, ...others] = peers
+  const [bob, oscar] = others
+
+  async function sync (label) {
+    return run(
+      label,
+      Promise.all(others.map(peer => replicate(alice, peer))),
+      { isTest: false }
+    )
+  }
+
+  await run(
+    'start tribes',
+    Promise.all(peers.map(peer => p(peer.tribes2.start)())),
+  )
+
+  const group = await run(
+    'alice creates a group',
+    p(alice.tribes2.create)({})
```

Comment:
:fire: avoid using promisify on tribes2 functions. it breaks stuff sometimes since they self-promisify

Reply:
GOOD TO KNOW! I will undo this